### DISCUSSION TITLE
How does the default "Buffer selection" inline assistant work?

### DISCUSSION THREAD

**Discussion by zhou13** [olimorris/codecompanion.nvim#537] [Q&A] [answered]
When I select several lines of code, press `<C-a>`, and then select "Buffer selection", the codecompanion.nvim does not ask me what I want to do with these lines of code.  Instead, it always replaces the buffer with something that does not make sense. What is the intended way to use the default "Buffer selection" inline assistant?

In addition, there is also a "Custom Prompt" inline assistant. But it seems that it just opens a chat. How could I command it to replace the buffer?

**Comment by olimorris** [comment, marked as answer]
> When I select several lines of code, press `<C-a>`, and then select "Buffer selection", the codecompanion.nvim does not ask me what I want to do with these lines of code. Instead, it always replaces the buffer with something that does not make sense. What is the intended way to use the default "Buffer selection" inline assistant?

So the buffer selection, as per its description (_"Send the current buffer to the LLM as part of an inline prompt"_) will send over all of the current buffer as context alongside your code selection and your code prompt:

![2024-12-15 10_16_01 - WezTerm](https://github.com/user-attachments/assets/2ff593c3-bf4f-4775-bca1-c00bc19f0fa9)

In the example above, I've selected some code in a mini.test file and asked the LLM to tell me what the selection does in the context of the whole buffer.

> In addition, there is also a "Custom Prompt" inline assistant. But it seems that it just opens a chat. How could I command it to replace the buffer?

As per the [Quickstart](https://github.com/olimorris/codecompanion.nvim#rocket-quickstart) guide, "the assistant will evaluate the prompt and either write code or open a chat buffer...". So it depends entirely on your prompt:

![2024-12-15 10_19_44 - WezTerm](https://github.com/user-attachments/assets/6e458eb4-c50e-4b47-b6d2-d18f791fd127)

![2024-12-15 10_20_52 - WezTerm](https://github.com/user-attachments/assets/6e951df4-91d0-4351-ae0c-76d4e4c57033)

**Comment by zhou13** [comment]
Thank you for the gifs. Now I can reproduce your behavior. For some reason, the prompt window does not show up yesterday for the "Buffer selection".
